design/reboot-interface.md

```diff
@@ -114,7 +126,14 @@ power off stage whenever any of the following occur:
 * The ``lastPoweredOn`` time is after the time of the reboot request
 * The ``poweredOn`` status is ``false``
 
-Power will then be restored only if/when the ``Online`` spec of the Host is true. 
+If the Host is deprovisioned, the controller automatically removes the
+``host.metal3.io/reboot`` annotation.
+
+In the case of multiple clients, clients detecting that another entity has
```

Comment:
I suddenly realise that in my head this whole time annotations were a _list_ of key-values pairs (so the same annotation could occur multiple times from multiple clients). But, in fact, they're not and it can't.
Would it make sense to use something like ``host.metal3.io/reboot/machine.openshift.io`` as the key, but have the controller treat any keys that begin with ``host.metal3.io/reboot/`` as a reboot annotation? (So it would ignore the last part, but allow multiple actors to interoperate at the same time.)

Otherwise if we don't add an annotation because there's an existing one there already, there's no guarantee that we will have deleted our Node by the time whatever added the annotation removes it again. (And if we do take it over then the other thing might have the same problem.) Allowing only one annotation and keeping the power off until you delete it is actually no better than just twiddling `Spec.Online`, except that we now have 2 of them.

Reply:
Ok, so this explains the disconnect with putting UIDs :-)

A single annotation containing a JSON blob with multiple annotations requires far too much trust.  So some sort of annotation prefix seems like the only option... unless we go back to a having a CRD or modify BMH to take a list of annotations or similar in the ``.Spec``.  But we'd need to be confident about this being a long term interface.

Obviously the controller has to wait for the list to be empty.